src/main/java/com/lpvs/entity/report/LPVSReportBuilder.java

```diff
@@ -248,6 +247,247 @@ public static void saveHTMLToFile(String htmlContent, String filePath) {
         }
     }
 
+    /**
+     * Generates a comment to the pull request for publication to the VCS.
+     *
+     * @param scanResults the results of the license scan
+     * @param conflicts a list of license conflicts found during the scan
+     * @param webhookConfig configuration related to the repository and webhook
+     * @param vcs the string representation of the version control system
+     * @return the code of the generated VCS comment
+     */
+    public String generatePullRequestComment(
+            List<LPVSFile> scanResults,
+            List<LPVSLicenseService.Conflict<String, String>> conflicts,
+            LPVSQueue webhookConfig,
+            LPVSVcs vcs) {
+
+        StringBuilder commitCommentBuilder = new StringBuilder();
+        commitCommentBuilder.append("**Detected Licenses:**\n\n");
+
+        Map<String, GroupInfo<?>> detectedLicenseInfo =
+                groupScanResultsForLicenseTable(scanResults);
+        long prohibitedLicenses = getDetectedLicenseCountByType(detectedLicenseInfo, prohibited);
+        long restrictedLicenses = getDetectedLicenseCountByType(detectedLicenseInfo, restricted);
+        long unreviewedLicenses = getDetectedLicenseCountByType(detectedLicenseInfo, unreviewed);
+        long licenseDetected = prohibitedLicenses + restrictedLicenses + unreviewedLicenses;
+
+        if (licenseDetected > 0) {
+            commitCommentBuilder.append("Potential license issues detected:\n");
+            if (prohibitedLicenses > 0) {
+                commitCommentBuilder.append(
+                        " - Prohibited license(s): " + prohibitedLicenses + "\n");
+            }
+            if (restrictedLicenses > 0) {
+                commitCommentBuilder.append(
+                        " - Restricted license(s): " + restrictedLicenses + "\n");
+            }
+            if (unreviewedLicenses > 0) {
+                commitCommentBuilder.append(
+                        " - Unreviewed license(s): " + unreviewedLicenses + "\n");
+            }
+        } else {
+            commitCommentBuilder.append("No license problems detected.\n");
+        }
+        if (scanResults != null && !scanResults.isEmpty()) {
+            commitCommentBuilder
+                    .append("\n")
+                    .append(generateLicenseTable(detectedLicenseInfo, webhookConfig, vcs))
+                    .append("\n");
+        }
```

Comment:
maybe add else { log.error ("empty scan result"}  ?

Reply:
It's normal when scanResults = null (no licenses detected: 95% of all scans have such a situation).